src/main/java/org/gbif/doi/service/datacite/RestJsonApiDataCiteService.java

```diff
@@ -0,0 +1,206 @@
+        dataCiteClient.createDoi(jsonApiWrapper);
+    }
+
+    /**
+     * Register a doi. Uses createDoi method with the event type 'PUBLISH'.
+     * @param doi the identifier
+     * @param metadata the metadata to be associated with the doi
+     */
+    @Override
+    public void register(DOI doi, URI target, DataCiteMetadata metadata) throws DoiException {
+        Preconditions.checkNotNull(doi);
+        Preconditions.checkNotNull(metadata);
+        String xmlMetadata = DataCiteValidator.toXml(doi, metadata);
+        register(doi, target, xmlMetadata);
+    }
+
+    /**
+     * Creates DoiSimplifiedModel which can be passed as an argument to dataCiteClient's create method.
+     * @param doi the identifier
+     * @param metadata the metadata to be associated with the doi
+     * @return doi model which can be registered or reserved
+     */
+    private DoiSimplifiedModel prepareDoiCreateModel(DOI doi, String metadata) {
+        DoiSimplifiedModel model = new DoiSimplifiedModel();
+        model.setId(doi.getDoiName());
+        model.setDoi(doi.getDoiName());
+        model.setXml(Base64.getEncoder().encodeToString(metadata.getBytes()));
+        return model;
+    }
+
+    /**
+     * Delete a doi.
+     * @param doi the identifier to delete
+     * @return true if successfully deleted, false otherwise
+     */
+    @Override
+    public boolean delete(DOI doi) {
+        Preconditions.checkNotNull(doi);
+        Response<Void> deleteResponse = dataCiteClient.deleteDoi(doi.getDoiName());
+        return deleteResponse.code() == 204;
+    }
+
+    /**
+     * Update with metadata.
+     * @param doi the identifier
+     * @param metadata the datacite metadata
+     */
+    @Override
+    public void update(DOI doi, String metadata) {
+        Preconditions.checkNotNull(doi);
+        Preconditions.checkNotNull(metadata);
+        DoiSimplifiedModel model = new DoiSimplifiedModel();
+        model.setXml(Base64.getEncoder().encodeToString(metadata.getBytes()));
+        JSONAPIDocument<DoiSimplifiedModel> jsonApiWrapper = new JSONAPIDocument<>(model);
+        dataCiteClient.updateDoi(doi.getDoiName(), jsonApiWrapper);
+    }
+
+    /**
+     * Update with metadata.
+     * @param doi the identifier
+     * @param metadata the datacite metadata
+     *
+     * @throws DoiException if some problems occur while xml serializing
+     */
+    @Override
+    public void update(DOI doi, DataCiteMetadata metadata) throws DoiException {
+        Preconditions.checkNotNull(doi);
+        Preconditions.checkNotNull(metadata);
+        String xmlMetadata = DataCiteValidator.toXml(doi, metadata);
+        update(doi, xmlMetadata);
+    }
+
+    /**
+     * Not implemented
+     * @param doi the identifier of metadata to update
+     * @param target the new URL the DOI should resolve to
+     */
+    @Override
+    public void update(DOI doi, URI target) {
+        throw new NotImplementedException("Method not implemented");
```

Comment:
shouldn't be UnsupportedOperationException?

Reply:
fixed